### ISSUE TITLE
The loss does not decreased and stationary

### ISSUE THREAD

**Issue by yihao31** [zhunzhong07/IDE-baseline-Market-1501#2]
when I trained with ResNet model in Market-1501,the loss was start from 87.3365 and stayed at it until  finished the training. but when use CaffeNet model ,the training process is ok and the loss is decrease. So i wonder is there any problem in ResNet model ? 
here is a part of training log:
`I0901 18:00:02.961964 23096 sgd_solver.cpp:106] Iteration 2380, lr = 0.01
I0901 18:00:08.045089 23096 solver.cpp:228] Iteration 2400, loss = 87.3365
I0901 18:00:08.045128 23096 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0901 18:00:08.045135 23096 sgd_solver.cpp:106] Iteration 2400, lr = 0.01
I0901 18:00:13.134079 23096 solver.cpp:228] Iteration 2420, loss = 87.3365
I0901 18:00:13.134107 23096 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0901 18:00:13.134114 23096 sgd_solver.cpp:106] Iteration 2420, lr = 0.01
I0901 18:00:18.219985 23096 solver.cpp:228] Iteration 2440, loss = 87.3365
I0901 18:00:18.220024 23096 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0901 18:00:18.220031 23096 sgd_solver.cpp:106] Iteration 2440, lr = 0.01
I0901 18:00:23.304426 23096 solver.cpp:228] Iteration 2460, loss = 87.3365
I0901 18:00:23.304466 23096 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0901 18:00:23.304471 23096 sgd_solver.cpp:106] Iteration 2460, lr = 0.01
I0901 18:00:28.420131 23096 solver.cpp:228] Iteration 2480, loss = 87.3365
I0901 18:00:28.420162 23096 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0901 18:00:28.420169 23096 sgd_solver.cpp:106] Iteration 2480, lr = 0.01
`

**Comment by zhunzhong07** [comment]
you should start the lr with 0.001, and fine-tuning from the model trained on ImageNet

**Comment by zhunzhong07** [comment]
@yihao31 you should start the lr with 0.001, and fine-tuning from the model pre-trained on ImageNet

**Comment by yihao31** [comment]
I set the lr as 0.001 and fine-tuning from the model pre-trained it doesn't work so I changed lr. 
here is my solver:
```
net: "models/market/ResNet_50/ResNet_50_train_val.prototxt"
test_iter: 10
test_interval: 500
test_initialization: false
display: 20
average_loss: 20
base_lr: 0.001
lr_policy: "step"
gamma: 0.01
stepsize: 20000
max_iter: 50000
momentum: 0.9
weight_decay: 0.0005
snapshot: 0
snapshot_prefix: "output/market_train/IDE_ResNet_50"
solver_mode: GPU 
```
and the log of training is as below:
```
Attempting to upgrade batch norm layers using deprecated params: data/imagenet_models//ResNet_50.caffemodel
I0902 17:51:31.226007 30672 upgrade_proto.cpp:80] Successfully upgraded batch norm layers using deprecated params.
I0902 17:51:31.243816 30672 caffe.cpp:251] Starting Optimization
I0902 17:51:31.243845 30672 solver.cpp:279] Solving Res50-base
I0902 17:51:31.243850 30672 solver.cpp:280] Learning Rate Policy: step
I0902 17:51:31.659909 30672 solver.cpp:228] Iteration 0, loss = 87.3365
I0902 17:51:31.659943 30672 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0902 17:51:31.659970 30672 sgd_solver.cpp:106] Iteration 0, lr = 0.001
I0902 17:51:36.666144 30672 solver.cpp:228] Iteration 20, loss = 87.3365
I0902 17:51:36.666173 30672 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0902 17:51:36.666182 30672 sgd_solver.cpp:106] Iteration 20, lr = 0.001
I0902 17:51:41.757443 30672 solver.cpp:228] Iteration 40, loss = 87.3365
I0902 17:51:41.757473 30672 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0902 17:51:41.757483 30672 sgd_solver.cpp:106] Iteration 40, lr = 0.001
I0902 17:51:46.852303 30672 solver.cpp:228] Iteration 60, loss = 87.3365
I0902 17:51:46.852342 30672 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
I0902 17:51:46.852351 30672 sgd_solver.cpp:106] Iteration 60, lr = 0.001
I0902 17:51:51.946357 30672 solver.cpp:228] Iteration 80, loss = 87.3365
I0902 17:51:51.946388 30672 solver.cpp:244]     Train net output #0: loss = 87.3365 (* 1 = 87.3365 loss)
```

**Comment by zhunzhong07** [comment]
@yihao31 did you use the same train_val.prototxt as I provided? It seems that the softmax loss is wrong.

**Comment by yihao31** [comment]
yes, I used the same tarin_val.prototxt. Can you run it again and check is there the same problem.

**Comment by zhunzhong07** [comment]
@yihao31 I have checked it, it works well for me and others. You can try to re-compile your caffe with comment USE_CUDNN := 1

**Comment by yihao31** [comment]
I did what you suggested and It got the same result. And I cloned your Re-Ranking code,  when trained IDE model, it appears same problem.

**Comment by yihao31** [comment]
if you can run it with no problem, may something wrong with my environment. thank you for answering me.
